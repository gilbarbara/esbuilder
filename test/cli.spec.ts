import { spawnSync } from 'child_process';

describe('cli', () => {
  beforeAll(() => {
    spawnSync('npm', ['run', 'build']);
    // jest.spyOn(console, 'log').mockImplementation(() => undefined);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should show the help', () => {
    const { stdout } = spawnSync('ts-node', ['./src/index.ts']);

    expect(stdout.toString()).toMatchSnapshot();
  });

  it('should run the CommonJS build', () => {
    const { stdout } = spawnSync('ts-node', [
      './src/index.ts',
      '--cjs',
      '--input=./test/__fixtures__/index.ts',
      '--outDirectory=./.tmp/dist',
    ]);

    expect(stdout.toString()).toMatchSnapshot();
  });

  it('should run the ESM build', () => {
    const { stdout } = spawnSync('ts-node', [
      './src/index.ts',
      '--esm',
      '--input=./test/__fixtures__/index.ts',
      '--outDirectory=./.tmp/dist',
    ]);

    expect(stdout.toString()).toMatchSnapshot();
  });
});
