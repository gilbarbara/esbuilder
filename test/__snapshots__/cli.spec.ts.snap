// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should run the CommonJS build 1`] = `
"[34mBuilding CommonJS...[39m

  .tmp/dist/index.js             1.6kb  100.0%
   â”” test/__fixtures__/index.ts  402b    24.1%

[32mâœ“ CommonJS build complete.[39m
"
`;

exports[`cli should run the ESM build 1`] = `
"[34mBuilding ESM...[39m

  .tmp/dist/index.mjs            302b   100.0%
   â”” test/__fixtures__/index.ts  201b    66.6%

[32mâœ“ ESM build complete.[39m
"
`;

exports[`cli should show the help 1`] = `
"Builds the source code into CommonJS and/or ESM modules.
    
Targets:
  --cjs            Build CommonJS module
  --esm            Build ESM module

Build Options:
  --input          Input files (default: ./src/index.ts)
  --externals      Add external packages to the bundle
  --noBundle       Skip bundling dependencies
  --noSourcemap    Skip the sourcemap
  --outDirectory   Output directory (default: ./dist)
  --platform       The platform (default: node)
  --target         The target environment (default: node14)

"
`;
